KomaMRIBase/src/datatypes/phantom/motion/SimpleMotion.jl

```diff
@@ -0,0 +1,53 @@
+# ------ SimpleMotionType
+abstract type SimpleMotionType{T<:Real} end
+
+"""
+Simple Motion
+
+x = x + ux
+y = y + uy
+z = z + uz
+
+"""
+# -------- SimpleMotion
+mutable struct SimpleMotion{S <: SimpleMotionType} <: MotionModel
+    types::AbstractVector{S}
+end
+
+Base.getindex(motion::SimpleMotion, p::Union{AbstractRange,AbstractVector,Colon}) = motion
+Base.getindex(motion::SimpleMotion, p::Union{AbstractRange,AbstractVector,Colon}, 
+                                    q::Union{AbstractRange,AbstractVector,Colon}) = motion
+
+function get_spin_coords(motion::SimpleMotion{S}, x::AbstractVector{T}, y::AbstractVector{T}, z::AbstractVector{T}, t::AbstractArray{T}) where {T<:Real, S<:SimpleMotionType{T}}
+    xt = x .+ reduce(.+, map((type) -> displacement_x(type, x, y, z, t), motion.types)) # Periodicity could be included here
+    yt = y .+ reduce(.+, map((type) -> displacement_y(type, x, y, z, t), motion.types))
+    zt = z .+ reduce(.+, map((type) -> displacement_z(type, x, y, z, t), motion.types))
+    return xt, yt, zt
+end
```

Comment:
Have you tried combining SimpleMotion types here? I think you are forcing it to all be the same.

Reply:
Yes, and it works as expected. For example:
`phantom.motion = SimpleMotion([Rotation(ti=0.2, tf=1.2, yaw=π/2, roll=π/4),
                                Translation(ti=0.2, tf=1.2, dx=0.03)])`

This combines two different SimpleMotionType's